app/src/main/java/ch/epfl/cs311/wanderwave/model/spotify/SpotifyController.kt

```diff
@@ -99,6 +101,87 @@ class SpotifyController(private val context: Context) {
     }
   }
 
+  /**
+   * Get all the playlist, title, ... from spotify from the home page of the user.
+   *
+   * @return a Flow of ListItem which has all the playlist, title, ... from the home page of the
+   *   user.
+   * @author Menzo Bouaissi
+   * @since 2.0
+   * @last update 2.0
+   */
+  fun getAllElementFromSpotify(): Flow<List<ListItem>> {
+    val list: MutableList<ListItem> = emptyList<ListItem>().toMutableList()
+    return callbackFlow {
+      val callResult =
+          appRemote?.let { it ->
+            it.contentApi
+                .getRecommendedContentItems(ContentApi.ContentType.DEFAULT)
+                .setResultCallback {
+                  for (i in it.items) list += i
+                  trySend(list)
+                }
+                .setErrorCallback { trySend(list + ListItem("", "", null, "", "", false, false)) }
+          }
+      awaitClose { callResult?.cancel() }
+    }
+  }
+  /**
+   * Get the children of a ListItem. In our case, the children is either a playlist or an album
+   *
+   * @param listItem the ListItem to get the children from
+   * @return a Flow of ListItem
+   * @author Menzo Bouaissi
+   * @since 2.0
+   * @last update 2.0
+   */
+  @OptIn(FlowPreview::class)
+  fun getChildren(listItem: ListItem): Flow<ListItem> {
+
+    return callbackFlow {
+      val callResult =
+          appRemote?.let { it ->
+            it.contentApi
+                .getChildrenOfItem(listItem, 50, 0)
+                .setResultCallback {
+                  for (i in it.items) if (i.id.contains("album") || i.id.contains("playlist"))
+                      trySend(i)
+                }
+                .setErrorCallback { trySend(ListItem("", "", null, "", "", false, false)) }
+          }
```

Comment:
Same here, why empty track and not just 'nothing?

Reply:
I don't know what you mean by nothing, the function expect as a result a Flow<ListItem>, so i have to return a flow of list item, i can't return a flow of null. Maybe there is a way to return "nothing", but i don't see how

Reply:
Yes it's what I meant return null and change the return type of the function to <ListItem? >. Someone made the same comment with one of my pr and I thought it was a good idea when dealing with async data, go deal with the nulls and errors in the view models or uis and not in the connection functions.